pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -389,14 +391,29 @@ func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]
 	if template == nil {
 		return nil, fmt.Errorf("no %q template defined", templateName)
 	}
+	proxyConfig := cfg.MeshConfig.GetDefaultConfig()
+	if d.env.PushContext != nil && d.env.PushContext.ProxyConfigs != nil {
+		if generatedProxyConfig := d.env.PushContext.ProxyConfigs.EffectiveProxyConfig(
+			&model.NodeMetadata{
+				Namespace: mi.Namespace,
+				// We need deployment labels here, so we have to check if the deployment exists and if so, get its labels.
+				// However, it won't exist on first creation of a gateway, so we might fall back
+				// to the default labels from the template: service.istio.io/canonical-name, istio.io/gateway-name.
+				Labels: mi.Labels,
+				// Not supported?
+				// Annotations: mi.Annotations,
```

Comment:
@howardjohn can you share your thoughts about handling labels and annotations?

Reply:
I think since we support it for everything else might as well...?

Reply:
Ok, I wanted to be sure, because in the issue you said "I would say we may consider intentionally not handling it for annotations".